Make snapshot_identifier computed (#18013)
hashicorp/terraform-provider-aws: .changelog/18013.txt

```release-note:enhancement

> resource/aws_db_instance: Allow `snapshot_identifier` to be removed from configuration by making it `computed`

Reviewer: Nit: Not sure if practitioners need to know about Computed

```suggestion
resource/aws_db_instance: Allow `snapshot_identifier` to be removed from configuration without resource recreation
```
